assets/src/stories-editor/components/block-mover/block-draggable.js

```diff
@@ -54,13 +90,108 @@ BlockDraggable.propTypes = {
 	children: PropTypes.any.isRequired,
 	onDragStart: PropTypes.func,
 	onDragEnd: PropTypes.func,
+	onNeighborHover: PropTypes.func,
+	onNeighborDrop: PropTypes.func,
+	getNeighborPageId: PropTypes.func,
 };
 
-export default withSelect( ( select, { clientId } ) => {
-	const { getBlockIndex, getBlockRootClientId } = select( 'core/block-editor' );
-	const rootClientId = getBlockRootClientId( clientId );
-	return {
-		index: getBlockIndex( clientId, rootClientId ),
-		rootClientId,
-	};
-} )( BlockDraggable );
+export default compose(
+	withSelect( ( select, { clientId } ) => {
+		const { getBlockIndex, getBlockRootClientId, getBlockOrder, getBlock } = select( 'core/block-editor' );
+		const rootClientId = getBlockRootClientId( clientId );
+		return {
+			index: getBlockIndex( clientId, rootClientId ),
+			rootClientId,
+			block: getBlock( clientId ),
+			getBlockOrder,
+		};
+	} ),
+	withDispatch( ( dispatch, { getBlockOrder, rootClientId, clientId, block, blockName } ) => {
+		const { setCurrentPage } = dispatch( 'amp/story' );
+		const { selectBlock, removeBlock, insertBlock, updateBlockAttributes } = dispatch( 'core/block-editor' );
+
+		const isCTABlock = 'amp/amp-story-cta' === blockName;
```

Comment:
Attachment and CTA, they are only one allowed per page. I can't see the validation here doesn't allow you to move these blocks to a page that already has one.

Reply:
Attachment isn't wrapped in Draggable, so it's not relevant here.

Reply:
CTA isn't not draggable either it seems. So why is this check required?

Reply:
Worth noting that the button within the CTA block is draggable. In case that makes a difference here.